client/src/components/CountdownTimer.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+const CountdownTimer: React.FC = () => {
+  const [timeLeft, setTimeLeft] = useState<number>(60);
+  const [isActive, setIsActive] = useState<boolean>(false);
+  const intervalRef = useRef<number | null>(null);
```

Comment:
You are currently managing two states: timeLeft and isActive. Consider if you could combine these into a single state object to reduce unnecessary re-renders, as each setState call triggers a re-render.